diag_manager/diag_manager.F90

```diff
@@ -4480,6 +4480,29 @@ SUBROUTINE diag_field_add_cell_measures(diag_field_id, area, volume)
     END IF
   END SUBROUTINE diag_field_add_cell_measures
 
+  !> @brief Copies a 3d buffer to a 4d buffer
+  !> @return a 4d buffer
+  function copy_3d_to_4d(data_in) &
+    result(data_out)
+    class (*), intent(in) :: data_in(:,:,:) !< Data to copy
+    class (*), allocatable :: data_out(:,:,:,:)
+
+    select type(data_in)
+    type is (real(kind=r8_kind))
+      allocate(real(kind=r8_kind) :: data_out(1:size(data_in,1), 1:size(data_in,2), 1:size(data_in,3), 1))
+      select type (data_out)
+      type is (real(kind=r8_kind))
+        data_out(:,:,:,1) = data_in
+      end select
+    type is (real(kind=r4_kind))
+      allocate(real(kind=r4_kind) :: data_out(1:size(data_in,1), 1:size(data_in,2), 1:size(data_in,3), 1))
+      select type (data_out)
+      type is (real(kind=r4_kind))
+        data_out(:,:,:,1) = data_in
+      end select
+    end select
```

Comment:
I think there should be some type of error check for types here, otherwise it could silently fail.

Reply:
https://github.com/NOAA-GFDL/FMS/pull/1347/commits/be486f72f223f6c14379557ea8abaa55c761929e